mozilla/fxa
# feat(react): Account recovery confirm key functionality

Will fill out the commit message and leave self-review when finished, I'm still finishing up tests.

--

How to test:
* Create an account, create a recovery key, sign out
* Go to either http://localhost:3030/?forceExperiment=generalizedReactApp&forceExperimentGroup=react and enter your email and then click “Reset Password,” or go to http://localhost:3030/reset_password?showReactApp=true
* Enter email and check inbox for email link
* Paste the link into your browser with `&showReactApp=true` parameters (we have a file up filed for this).
* You’ll be taken to `account_recovery_confirm_key`. You can also try hitting the email link without a recovery key
* Test as needed (enter bogus key, real key etc.). Note if you wait too long the link will expire and you’ll be taken to the damaged link screen. If successful you’ll be taken to `account_recovery_reset_password` (separate ticket)

Check the storybook link for states.

Here's a new error screen, error copy confirmed by Vesta:
<img src="https://user-images.githubusercontent.com/13018240/221685418-a78ad69e-445c-4823-b531-9bde96c0ed39.png" width="400px">
